### DISCUSSION TITLE
How can I convert a CDF_CHAR*24 variable into a CDF_TIME_TT2000?

### DISCUSSION THREAD

**Discussion by luisagvarruda** [spacepy/spacepy#701] [Support and Q&A] [answered]
My issue is ...

### Minimal example to reproduce issue:

```
Sample code to reproduce the problem
```

### Error message/Traceback:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  ...
```

**Comment by drsteve** [comment, marked as answer]
As this is a "how to" question I've converted to a discussion.

There's not really enough information given here to help with your exact problem.
However, given the data type `CDF_CHAR*24` I surmise that it's a string variable. Is it, perhaps, in the format "2001-01-01T12:30:00.000Z"? That'd give a variable of the right length...
If so the issue is that strings aren't auto-parsed in `pycdf` to see whether or not they're times.

In which case you should convert to timezone-naive datetimes (the trailing Z implies Zulu time, aka UTC) and then `pycdf` will convert to TT2000:

```
>>> import numpy as np
>>> from spacepy import pycdf
>>> import spacepy.time as spt
>>> fh = pycdf.CDF('test.cdf', '')
>>> fh['test'] = ['2014-01-01T12:00:00.333Z']
>>> fh['test_epoch'] = spt.Ticktock(np.asarray(fh['test']).astype('<U23')).UTC
>>> fh
<CDF:
test: CDF_CHAR*24 [1]
test_epoch: CDF_TIME_TT2000 [1]
>
```

**Reply by jtniehof** [reply]
This is basically it. I'll add that addressing #698 would simplify this somewhat.

Changing the type of a variable isn't possible once it's been written to or if the new type isn't compatible with the old. CDF_TIME_TT2000 is not compatible with CDF_CHAR*24.
